src/main/java/hotelreservation/domain/Room.java

```diff
@@ -0,0 +1,61 @@
+package hotelreservation.domain;
+import javax.persistence.*;
+import java.util.Date;
+
+@Entity
+@Table(name = "rooms")
+public class Room {
+
+    @Id
+    @GeneratedValue( strategy=GenerationType.AUTO )
+    private int idrooms;
+    private Date date;
+    private int free_rooms;
+
+
+    @ManyToOne
+    private Hotel hotel;
+
+    public Room() {
+    }
+
+    public Room(int idrooms, Date date, int free_rooms, Hotel hotel) {
+        this.idrooms = idrooms;
```

Comment:
Im not sure we need to care about overall # of rooms. It will be nice for displaying 10/90 rooms or whatever but not strictly needed for reservation.

Reply:
we could create a view or something if you want to go that way, but in order to do a full query you need to be able to query for available rooms in the hotel.  happy to discuss more.